tests/unit/pubsub_v1/opentelemetry/test_opentelemetry.py

```diff
@@ -0,0 +1,97 @@
+import mock
+import json
+import sys
+import types as stdlib_types
+import unittest
+from importlib import reload
+
+from google.auth import credentials
+from google.api_core import bidi
+from google.cloud.pubsub_v1 import types
+from google.cloud.pubsub_v1 import publisher
+from google.cloud.pubsub_v1 import subscriber
+from google.cloud.pubsub_v1 import opentelemetry_tracing
+
+from opentelemetry import trace
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import SimpleExportSpanProcessor
+from opentelemetry.sdk.trace.export.in_memory_span_exporter import InMemorySpanExporter
+
+TEST_SPAN_NAME = "foo"
+
+TEST_SPAN_ATTRIBUTES = {"foo": "bar"}
+
+TEST_PARENT_SPAN_CONTEXT = {
+    "trace_id": 253178191101418583724305005272987492287,
+    "span_id": 17783848543062801337,
+    "is_remote": False,
+    "trace_flags": 1,
+    "trace_state": {},
+}
+
+
+def _create_publisher():
+    creds = mock.Mock(spec=credentials)
+    return publisher.Client(credentials=creds)
+
+
+def _create_subscriber():
+    creds = mock.Mock(spec=credentials)
+    return subscriber.Client(credentials=creds)
+
+
+def _simulate_pubsub_session():
+    test_publisher = _create_publisher()
+    test_subscriber = _create_subscriber()
+    test_subscriber.subscribe("test_topic", callback=mock.sentinel.callback)
+    test_publisher.publish("test_topic", b"foobar")
+
+
+def _convert_context_to_dict(context):
+    return {
+        "trace_id": context.trace_id,
+        "span_id": context.span_id,
+        "is_remote": context.is_remote,
+        "trace_flags": context.trace_flags,
+        "trace_state": context.trace_state,
+    }
+
+
+class TestOpenTelemetry(unittest.TestCase):
```

Comment:
Recommendation - most (all?) of the existing unit tests use the `pytest` framework, thus let's be consistent and use it in these tests, too. Should make them more concise with less boilerplate. :wink:

Reply:
I used the unittest framework here because of the setup and teardown methods that make it easier to create the OpenTelemetry environment for the test cases. This can be done for each test in the `pytest` framework, but there would be some repeated code.

Reply:
I'm curious where `@pytest.fixture` falls short here? In should be possible to achieve the same setup / teardown behavior without code repetition and just pass the fixture as an argument to the tests that need it.

I'm not super strongly opinionated here, though, the unittest framework does the job fine, it's just that using a single test framework means somewhat lower maintenance cost in the long run.